Python/bytecodes.c

```diff
@@ -1553,6 +1554,41 @@ dummy_func(
             PREDICT(JUMP_BACKWARD);
         }
 
+        inst(LOAD_SUPER_ATTR, (global_super, class, self -- res2 if (oparg & 1), res)) {
+            PyObject *name = GETITEM(frame->f_code->co_names, oparg >> 2);
+            if (global_super == (PyObject *)&PySuper_Type && PyType_Check(class)) {
+                int meth_found = 0;
+                Py_DECREF(global_super);
+                res = _PySuper_Lookup((PyTypeObject *)class, self, name, oparg & 1 ? &meth_found : NULL);
+                Py_DECREF(class);
+                if (res == NULL) {
+                    Py_DECREF(self);
+                    ERROR_IF(true, error);
+                }
+                // Works with CALL, pushes two values: either `meth | self` or `NULL | meth`.
+                if (meth_found) {
+                    res2 = res;
+                    res = self;  // transfer ownership
+                } else {
+                    res2 = NULL;
+                    Py_DECREF(self);
+                }
+            } else {
+                PyObject *super;
+                if (oparg & 2) {
+                    super = PyObject_CallNoArgs(global_super);
+                } else {
+                    PyObject *stack[] = {class, self};
+                    super = PyObject_Vectorcall(global_super, stack, 2, NULL);
+                }
```

Comment:
Replace the above block with:
```C
    PyObject *stack[] = {class, self};
    super = PyObject_Vectorcall(global_super, stack, 2-(oparg&2), NULL);
```

If you flip the sense of the `oparg&2` bit, you can skip the subtraction.

Reply:
Great suggestion, thanks!